pkg/component/types.go

```diff
@@ -71,3 +74,63 @@ const (
 	// StateTypeUnknown means that odo cannot tell its state
 	StateTypeUnknown State = "Unknown"
 )
+
+func newComponentWithType(componentName, componentType string) Component {
+	cmp := NewComponent(componentName)
+	cmp.Spec.Type = componentType
+	return cmp
+}
+
+// NewComponent provides a constructor to component struct with some metadata prefilled
+func NewComponent(componentName string) Component {
+	return Component{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       ComponentKind,
+			APIVersion: machineoutput.APIVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: componentName,
+		},
+		Status: ComponentStatus{},
+	}
+}
+
+// newComponentList returns list of devfile and s2i components in machine readable format
+func newComponentList(s2iComps []Component) ComponentList {
```

Comment:
If this is for both then why use name s2iComps?

Reply:
Good catch, fixed